controllers/meshgateway_controller.go

```diff
@@ -126,7 +126,7 @@ func (r *IstioMeshGatewayReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return istiomeshgateway.NewChartReconciler(helmReconciler, servicemeshv1alpha1.IstioMeshGatewayProperties{
 			Revision:                fmt.Sprintf("%s.%s", icp.GetName(), icp.GetNamespace()),
 			EnablePrometheusMerge:   utils.BoolPointer(enablePrometheusMerge),
-			InjectionTemplate:       "gateway",
+			InjectionTemplate:       "gateway, gatewayOverrides",
```

Comment:
Not very important, but I'm thinking if it would be better that in case there is no `gatewayOverrides` (or `sidecarOverrides`) provided through the ICP spec, then it might be more optimal to have only `gateway` (or `sidecar`) as the injection template as usual.

For a few reasons:
- I know it's an empty spec merge, but still no need to do that template merge on the Istio side for the overrides
- It might be confusing to the user why that overrides is there if it's specified nowhere by the user in a normal installation (without providing overrides in the ICP spec)
- Also, if there is a bug in our implementation at any time in the future this solution won't mess up default installations without these overrides

What do you think?